modules/tensor_mechanics/include/userobjects/CrystalPlasticityStateVariableGSS.h

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef CRYSTALPLASTICITYSTATEVARIABLEGSS_H
+#define CRYSTALPLASTICITYSTATEVARIABLEGSS_H
+
```

Comment:
Do you need a separate class for state variable update for the phenomenological model since this class primarily adds the individual increments of internal variables.